### DISCUSSION TITLE
The specified executable is not a valid application for this OS platform

### DISCUSSION THREAD

**Discussion by DaddyXtof** [xmake-io/xmake#4404] [General]
Hi,

Under Windows 11 I have installed xmake using MSYS2:
``` pacman -Sy mingw-w64-x86_64-xmake ```
but get:
``` Program 'xmake.exe' failed to run: The specified executable is not a valid application for this OS platform.At line:1 char:1 ```
... when trying to run it.
Seems like a basic issue (may be bitness) but can't figure out what I am doing wrong in the install. This should all be 64 bits.
Anyone seen this before?

Thank you

**Comment by waruqi** [comment]
install failed or run failed?

if installation is success, you can try run `xmake` without `.exe` suffix. 

because it's bat script `xmake.bat` in bin directory.

**Reply by DaddyXtof** [reply]
**xmake.bat** is not available anywhere and not installed as part of the mingw package.
```└─Δ where.exe xmake.bat```
```INFO: Could not find files for the given pattern(s).```

The pacman install completes without errors.
```[2023-11-20T23:19:01+0000] [PACMAN] Running 'pacman -Sy mingw-w64-x86_64-xmake'
[2023-11-20T23:19:01+0000] [PACMAN] synchronizing package lists
[2023-11-20T23:19:25+0000] [ALPM] transaction started
[2023-11-20T23:19:29+0000] [ALPM] reinstalled mingw-w64-x86_64-xmake (2.8.5-1)
[2023-11-20T23:19:29+0000] [ALPM] transaction completed

**Comment by waruqi** [comment]
It works for me.

```console
:: Processing package changes...
(1/1) reinstalling mingw-w64-x86_64-xmake                                                                                                                          [####################################################################################################] 100%

wangrunqing@IIGS-A8825-D MINGW64 /d/projects/personal/xmake
$ xmake --version
xmake v2.8.5+20231107, A cross-platform build utility based on Lua
Copyright (C) 2015-present Ruki Wang, tboox.org, xmake.io
                         _
    __  ___ __  __  __ _| | ______
    \ \/ / |  \/  |/ _  | |/ / __ \
     >  <  | \__/ | /_| |   <  ___/
    /_/\_\_|_|  |_|\__ \|_|\_\____|
                         by ruki, xmake.io

    👉  Manual: https://xmake.io/#/getting_started
    🙏  Donate: https://xmake.io/#/sponsor
```

It should bin/xmake.exe, but it's a sh script.

```console
wangrunqing@IIGS-A8825-D MINGW64 /d/projects/personal/xmake
$ which xmake
/mingw64/bin/xmake

wangrunqing@IIGS-A8825-D MINGW64 /d/projects/personal/xmake
$ cat /mingw64/bin/xmake.exe
#!/usr/bin/env bash
BASEDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
if [ -f "$BASEDIR/../share/xmake/xmake.exe" ]; then
    $BASEDIR/../share/xmake/xmake.exe "$@"
fi
```

**Reply by waruqi** [reply]
But it does have some issues here as well, and I'll improve it, but that shouldn't affect the use of it right now. https://github.com/xmake-io/xmake/issues/3982#issuecomment-1822162208

**Reply by DaddyXtof** [reply]
FYI: I have tried on a separate Windows 11 machine where xmake was installed (via MSYS2/MinGW and pacman) and working well.
Doing an update (pacman -Syu) breaks xmake and then gives:
```
Program 'xmake.exe' failed to run: The specified executable is not a valid application for this OS platform.
```
So I believe the latest **mingw-w64-x86_64-xmake** package does not provide the right binaries.
